src/server/system_services/account_server.js

```diff
@@ -299,6 +299,48 @@ async function generate_account_keys(req) {
     });
 }
 
+/**
+ *
+ * UPDATE_ACCOUNT_KEYS
+ *
+ */
+async function update_account_keys(req) {
+    const account = system_store.get_account_by_email(req.rpc_params.email);
+    if (!account) {
+        throw new RpcError('NO_SUCH_ACCOUNT', 'No such account email: ' + req.rpc_params.email);
+    }
+    if (req.system && req.account) {
+        if (!is_support_or_admin_or_me(req.system, req.account, account)) {
+            throw new RpcError('UNAUTHORIZED', 'Cannot update account');
+        }
+    }
+    if (account.is_support) {
+        throw new RpcError('FORBIDDEN', 'Cannot update support account');
+    }
+    const access_keys = req.rpc_params.access_keys;
+
+    access_keys.secret_key = system_store.master_key_manager.encrypt_sensitive_string_with_master_key_id(
+        access_keys.secret_key, account.master_key_id._id);
+
+    return system_store.make_changes({
+        update: {
+            accounts: [{
+                _id: account._id,
+                access_keys: [
+                    access_keys
+                ]
+            }]
+        }
+    })
+    .then(() => Dispatcher.instance().activity({
```

Comment:
In which cases do we post events in the Dispatcher and where can we see them?
Can we write it in `async/await` code style instead of promise chaining `.then(`?

Reply:
Not sure about other things, but we can see those logs inside `noobaa-core-0` pod in `core` container.
For example(tried it for generate_credentials):
```$ kubectl logs noobaa-core-0 -c core| grep generate_credentials
May-13 13:34:05.669 [WebServer/38]    [L0] core.server.notifications.dispatcher:: Adding ActivityLog entry { event: 'account.generate_credentials', level: 'info', system: 6642162b2aeb8e0026fcec53, actor: 6642162d2aeb8e0026fcec62, account: 664216592aeb8e0026fcec6b, desc: SENSITIVE-d0a52793ea7cebdb }```